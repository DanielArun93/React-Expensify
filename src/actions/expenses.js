import uuid from 'uuid';
import database from '../firebase/firebase';

//Add expenses action generators
export const addExpense = (expense) => {
    return {
        type: 'ADDEXPENSES',
        expense
    }
}

export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
            description = '',
            note = '',
            amount = 0,
            createdAt = 0
        } = expenseData

        const expense = { description, note, amount, createdAt };
        return database.ref('expenses').push(expense).then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }))
        })

    }
}

//Remove expense
export const removeExpense = ({id} = {}) => {
    return {
        type: 'REMOVEEXPENSE',
        id
    }
}

//edit expense
export const editExpense = (id, update) => {
    return {
        type: 'EDITEXPENSE',
        id,
        update
    }
}