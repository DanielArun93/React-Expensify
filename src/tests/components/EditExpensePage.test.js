import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensepage/EditExpensepage';
import expenses from '../../fixtures/expenses';

let editExpense, removeExpense, history, wrapper, match;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage expense={expenses[0]} history={history} editExpense={editExpense} removeExpense={removeExpense} />)
})

test('should render editExpense correctly', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should handle on submit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[0].id, expenses[0]);
})

test('should handle on remove', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({
        id: expenses[0].id
    })
})