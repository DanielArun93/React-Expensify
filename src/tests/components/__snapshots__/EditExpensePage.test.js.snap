// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render editExpense correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 12000,
        "createdAt": 0,
        "description": "Rent",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
